### PR TITLE
Do not allow metrics.json to modify the MetricsTree

### PR DESCRIPTION
**Pull request by adleong**
Calling metrics.json with the q parameter for a subtree that does not exist
creates the subtree.  For example:

```
$ curl 'http://localhost:9990/admin/metrics.json?tree=1&q=jvm/mem/allocations/eden/bytes/foo/bar/bad'
{}
$ curl 'http://localhost:9990/admin/metrics.json?tree=1&q=jvm/mem/allocations/eden/bytes'
{"gauge":3.808624E8,"foo":{"bar":{"bad":{}}}}
```

We introduce `tryResolve` as an alternative to `resolve` that does not modify
the MetricsTree and use that when rendering metrics.json.

### CODE REVIEW

File: telemetry/admin-metrics-export/src/main/scala/io/buoyant/telemetry/admin/AdminMetricsExportTelemeter.scala [hunk 1]
```diff
@@ -33,17 +33,23 @@ class AdminMetricsExportTelemeter(
...
-      response.withOutputStream(writeFlatJson(_, subtree, pretty))
+    subtree match {
+      case Some(st) =>
+        if (tree)
+          response.withOutputStream(writeJsonTree(_, st))
+        else
+          response.withOutputStream(writeFlatJson(_, st, pretty))
+      case None =>
+        response.status = Status.NotFound
+        response.contentString = s""""No such subtree: $q""""
```
Comment (esbie) [inline telemetry/admin-metrics-export/src/main/scala/io/buoyant/telemetry/admin/AdminMetricsExportTelemeter.scala]: shouldn't this content string be a json object?

Comment (adleong) [inline telemetry/admin-metrics-export/src/main/scala/io/buoyant/telemetry/admin/AdminMetricsExportTelemeter.scala]: Maybe?  Currently it's a json string, but I'm open to suggestions.

Comment (esbie) [inline telemetry/admin-metrics-export/src/main/scala/io/buoyant/telemetry/admin/AdminMetricsExportTelemeter.scala]: I would like to suggest `{"error": "No such subtree: $q"}`

Comment (adleong) [inline telemetry/admin-metrics-export/src/main/scala/io/buoyant/telemetry/admin/AdminMetricsExportTelemeter.scala]: thank you! will update.

**Pull request merged**